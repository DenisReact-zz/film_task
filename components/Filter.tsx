import React, { FC } from "react";
import s from "../styles/Filter.module.css";

interface PropsFilter {
  onChangeFilter: (str: string) => void;
}
const Filter: FC<PropsFilter> = ({ onChangeFilter }) => {
  const thumb = React.useRef(null);

  const click_button = (filter: string, index: number) => () => {
    onChangeFilter(filter);
    thumb.current.style.left = `${50 * index}%`;
  };
  return (
    <div className={s.filter} data-testid="filters">
      <div className={s.blocks}>
        <button
          onClick={click_button("descending", 0)}
          data-testid="buttonOfFilter"
        >
          Descending
        </button>
        <button
          onClick={click_button("ascending", 1)}
          data-testid="buttonOfFilter"
        >
          Ascending
        </button>
        <div className={s.thumb} ref={thumb}></div>
      </div>
    </div>
  );
};

export default Filter;
