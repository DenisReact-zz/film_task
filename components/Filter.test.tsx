import { render, screen } from "@testing-library/react";
import Filter from "./Filter";
describe("Tests for filters", () => {
  const mockFn = jest.fn();
  beforeEach(() => {
    render(<Filter onChangeFilter={mockFn} />);
  });
  it("check layout filters", () => {
    const blockFilters = screen.getByTestId("filters");
    expect(blockFilters).toMatchSnapshot();
  });
  it("check buttons in filters", () => {
    const buttonsFilters = screen.queryAllByTestId("buttonOfFilter");
    expect(buttonsFilters.length).toBeGreaterThan(1);
  });
});
