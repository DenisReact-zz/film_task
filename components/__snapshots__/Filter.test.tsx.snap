// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for filters check layout filters 1`] = `
<div
  class="filter"
  data-testid="filters"
>
  <div
    class="blocks"
  >
    <button
      data-testid="buttonOfFilter"
    >
      Descending
    </button>
    <button
      data-testid="buttonOfFilter"
    >
      Ascending
    </button>
    <div
      class="thumb"
    />
  </div>
</div>
`;
